x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsConstants.java

```diff
@@ -30,6 +30,8 @@
 
     public static final String SNAPSHOT_DIRECTORY_FACTORY_KEY = "snapshot";
 
+    public static final String SNAPSHOT_RECOVERY_STATE_FACTORY_KEY = "on_demand";
```

Comment:
With the change in behavior of this PR, rename this perhaps to "snapshot_prewarm".